inst/include/Rcpp/vector/DimNameProxy.h

```diff
@@ -56,6 +56,31 @@ namespace internal{
             return *this;
         }
 
+        inline DimNameProxy& operator=(const DimNameProxy& other) {
+            SEXP dim = Rf_getAttrib(data_, R_DimSymbol);
+            if (INTEGER(dim)[dim_] != INTEGER(dim)[other.dim_]) {
+                std::stringstream s;
+                s << "dimension extent is '"
+                  << INTEGER(dim)[dim_]
+                  << "' while length of names is '"
+                  << INTEGER(dim)[other.dim_]
+                  << "'";
+                stop(s.str());
+            }
+            SEXP dimnames = Rf_getAttrib(data_, R_DimNamesSymbol);
+            SEXP dims = Rf_getAttrib(data_, R_DimSymbol);
```

Comment:
It looks like you already retrieved this above with `SEXP dim = Rf_getAttrib(data_, R_DimSymbol)` -- is this line superfluous?